File: API_MySIRH/Entities/Carriere.cs
namespace API_MySIRH.Entities
{
    public class Carriere : EntityBase
    {
        public virtual int? CollaborateurId { get; set; }
        public virtual Collaborateur? Collaborateur { get; set; }
        public virtual int? NiveauId { get; set; }
        public virtual Niveau? Niveau { get; set; }
        public virtual int? PosteId { get; set; }
        public virtual Post? Poste { get; set; }
        public string? ProfilDeCout { get; set; }
        public float? SalaireNet { get; set; }
        public float? VariableNet { get; set; }
        public float? SalaireBrut { get; set; }
        public float? VariableBrut { get; set; }
        public string? TLRH { get; set; }
    }
}

File: Client/src/app/Models/Carriere.ts
import { Collaborator } from "./Collaborator";
import { BaseMDM, Niveau, Poste } from "./MdmModel";

export class Carriere extends BaseMDM {
    id: number = 0;
    collaborateurId?: number = 0;
    collaborateur?: Collaborator;
    niveauId?: number = 0;
    niveau?: Niveau;
    posteId?: number = 0;
    poste?: Poste;
    profilDeCout?: string = "";
    salaireNet?: number = 0;
    variableNet?: number = 0;
    salaireBrut?: number = 0;
    variableBrut?: number = 0;
    tlrh?: string = "";
    annee: number = 0;
}